platform/default/layer_manager.cpp

```diff
@@ -0,0 +1,64 @@
+#include <mbgl/style/layer.hpp>
+#include <mbgl/style/layers/symbol_layer.hpp>
+#include <mbgl/style/layers/background_layer.hpp>
+#include <mbgl/style/layers/circle_layer.hpp>
+#include <mbgl/style/layers/fill_extrusion_layer.hpp>
+#include <mbgl/style/layers/fill_layer.hpp>
+#include <mbgl/style/layers/heatmap_layer.hpp>
+#include <mbgl/style/layers/hillshade_layer.hpp>
+#include <mbgl/style/layers/line_layer.hpp>
+#include <mbgl/style/layers/raster_layer.hpp>
+#include <mbgl/style/layers/symbol_layer.hpp>
+
+#include <memory>
+#include <vector>
+
+namespace mbgl {
+namespace style {
+ 
+class LayerManagerBase : public LayerManager {
+public:
+    LayerManagerBase();
+private:
+    // LayerManager overrides.
+    std::unique_ptr<Layer> createLayer(const std::string& type, const std::string& id, const conversion::Convertible& value, conversion::Error& error) noexcept final;
+    std::vector<std::unique_ptr<LayerFactory>> factories;
+};
+
+LayerManagerBase::LayerManagerBase() {
+    factories.emplace_back(std::unique_ptr<LayerFactory>(new FillLayerFactory));
```

Comment:
`factories.emplace_back(std::make_unique<FillLayerFactory>());` ?

Reply:
it would need cast to `std::unique_ptr<LayerFactory>`